### DISCUSSION TITLE
Initial admin user creation

### DISCUSSION THREAD

**Discussion by Hugo-Dz** [pocketbase/pocketbase#4489] [Q&A]
Hey!

Context:
I use Pocketbase alongside a SvelteKit app running in Docker containers.

Issue:
I wonder if there is any ways to create the very first admin user through my SvelteKit app or programmatically?

Goal:
Ship self-hosted apps where users only have to deal with the SvelteKit front-end and use Pocketbase as a backend

Cheers !

**Comment by ganigeorgiev** [comment]
You can create the first admin account:

1. Using the [Admin Create API](https://pocketbase.io/docs/api-admins/#create-admin) (the first admin account creation doesn't require authentication).

2. Using the `./pocketbase admin create test@example.com 1234567890` (_if no pb_data is available requires `./pocketbase migrate up` before that; after the [refactoring](https://github.com/pocketbase/pocketbase/discussions/4355) this no longer will be required_).

3. (_Recommended_) As part of a migration (migration are applied automatically on `serve`):
    - Go: https://pocketbase.io/docs/go-migrations/#creating-new-admin
    - JS `pb_migrations`: https://pocketbase.io/docs/js-migrations/#creating-new-admin

**Reply by Hugo-Dz** [reply]
Thank you for your quick reply!

I guess I need to use the 1st method as I want the user to create the first admin account through my SvelteKit front-end?

The 3rd method works well but It needs to be hardcoded in the `pb_migrations` file right? I can't use this method within one of my SveltKit pages to create the first user? (If I understand correctly)

[EDIT]: In the case of using the 1st method, what is the recommended way to check if there is a first admin account already created? This check will take place in a SvelteKit server load function.

**Reply by ganigeorgiev** [reply]
> The 3rd method works well but It needs to be hardcoded in the pb_migrations file right?
> I can't use this method within one of my SveltKit pages to create the first user?

Yes, in this case the admin will have to be hardcoded as a migration file and No - the JS code in `pb_hooks` and `pb_migrations` is executed only by PocketBase.

> In the case of using the 1st method, what is the recommended way to check if there is a first admin account already created?

In case there is already an admin account, the API call will fail with 401 because your request will lack an admin auth token, or in other words - you can try to wrap it in a `try/catch` and inspect the [`ClientResponseError` fields](https://github.com/pocketbase/js-sdk?tab=readme-ov-file#error-handling).

**Reply by Hugo-Dz** [reply]
Great, so here's how I managed the thing, let me know your thoughts:

The goal is to let the user create his admin account in a SvelteKit front-end after spinning up the machine (SvelteKit and Pocketbase packed in containers)

``` js
// +page.server.ts (index page)
// Using an hard-coded first dummy admin in the /pb_migrations
export const load: PageServerLoad = async (event) => {
	const pb = new PocketBase("http://nginx/pocket-base");
	let dummyAdmin;
	try {
		dummyAdmin = await pb.admins.authWithPassword("admin@admin.com", "1234567890");
	} catch (e) {
		// No dummy admin found, there are already a real admin in the system, continue to index page
		return {
			success: true
		}
	}
	if (dummyAdmin && dummyAdmin.admin) {
		// Dummy admin found, redirect to get-started page to update the dummy admin with user credentials
		redirect(301, "/get-started");
	}
};
```
